api/progress.go

```diff
@@ -176,8 +177,16 @@ func (r progressRoutes) markWatched(c *gin.Context) {
 }
 
 func (r progressRoutes) getProgressBatch(c *gin.Context) {
-	tumLiveContext := c.MustGet("TUMLiveContext").(tools.TUMLiveContext)
+	ctx, exists := c.Get("TUMLiveContext")
```

Comment:
Was there an issue with the `MustGet` approach?

Reply:
For the progress API we don't use a middleware as `tools.InitCourse(daoWrapper)` and therefore the handler panics every time the context doesn't exists instead of returning `400`.